Public/Src/Engine/Processes/SandboxedProcessPipExecutor.cs

```diff
@@ -673,6 +673,23 @@ public async Task<SandboxedProcessPipExecutionResult> RunAsync(CancellationToken
                     string arguments = m_pip.Arguments.ToString(m_pipDataRenderer);
                     m_timeout = GetEffectiveTimeout(m_pip.Timeout, m_sandboxConfig.DefaultTimeout, m_sandboxConfig.TimeoutMultiplier);
 
+                    // If ChangeAffectedInputListWrittenFilePath is set, write the change affected inputs of the pip to the file.
+                    if (m_pip.ChangeAffectedInputListWrittenFilePath != default)
+                    {
+                        var changeAffectedInputs = m_pip.GetChangeAffectedInputNames(m_pathTable);
+                        if (changeAffectedInputs != ReadOnlyArray<string>.Empty)
+                        {

```

Comment:
>{ [](start = 24, length = 1)

If you only write the file when the change inputs are non-empty, then the pip would fail if it expects change input file (by specifying some command line argument), but no such file is created by BuildXL.

Reply:
I will generate a file with empty content in this case. However, there are some other cases, for example removing file/directory, that I will not generate a file which indicate qtest to instrument all load files. I talked to Aniruth and he agree to add a check in qtest to see if the file(by specifying some command line argument) doesn't exist, qtest will instrument all.

Reply:
We should not make any specific assumption about QTest here.

---
In reply to: [314851206](https://github.com/microsoft/BuildXL/pull/748#discussion_r314851206) [](ancestors = 314851206)